waku/v2/protocol/waku_filter_v2/protocol.nim

```diff
@@ -0,0 +1,159 @@
+## Waku Filter protocol for subscribing and filtering messages
+
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+  std/[options,sequtils,sets,tables],
+  chronicles,
+  chronos,
+  libp2p/peerid,
+  libp2p/protocols/protocol
+import
+  ../../node/peer_manager,
+  ../waku_message,
+  ./common,
+  ./protocol_metrics,
+  ./rpc_codec,
+  ./rpc
+
+logScope:
+  topics = "waku filter"
+
+const
+  MaxSubscriptions* = 1000 # TODO make configurable
```

Comment:
perhaps a nitpick but

`MaxTotalSubscriptions`
`MaxPeerSubscriptions`

Reply:
I agree. Will change in next PR.